src/flowgraph/RampLinear.cpp

```diff
@@ -32,6 +32,10 @@ void RampLinear::setLengthInFrames(int32_t frames) {
 
 void RampLinear::setTarget(float target) {
     mTarget.store(target);
+    // If the ramp has not been used then start immediately at this level.
+    if (mLastCallCount < 0) {
```

Comment:
Could check == kInitialCallCount

Reply:
Good idea! Addressed in the second commit!